src/preload.py

```diff
@@ -65,6 +65,47 @@
 PARTITIONS_CACHE = {}
 
 
+class RetryCounter:
+    """Counter and logger for the number of times that a function is retried.
+    Usage:
+        retry_counter = RetryCounter()
+        hint = "Try reducing the system entropy"
+        wrapped_func, wrap_key = retry_counter.wrap(my_func, hint, *args)
+        retry_call(wrapped_func, fargs=args, ...)
+        retry_counter.clear(wrap_key)
+    """
+
+    def __init__(self):
+        self.counter = {}
+
+    @staticmethod
+    def key(*args):
+        return " ".join(str(e) for e in args)
+
+    def clear(self, key, *args):
+        del self.counter[key]
+
+    def inc(self, key, *args):
+        self.counter[key] = self.counter.setdefault(key, 0) + 1
+        return self.counter[key]
+
+    def wrap(self, func, hint, *args, **kwargs):
+        """Return a function that wraps the given func, counting its usage"""
+        key = self.key(*args)
+
+        def wrapped(*args, **kwargs):
+            count = self.inc(key, *args)
+            if count > 1:
+                log.info("\nRetrying (count={}) {} {}\n{}".format(
+                    count, func.__name__, key, hint))
```

Comment:
```suggestion
                log.info(
                    "\nRetrying (count={}) {} {}\n{}".format(
                        count,
                        func.__name__,
                        key,
                        hint,
                    ),
                )
```

Reply:
Done, plus one other small change to the `RetryCounter.key()` method: the function name was added to the `key` computation (previously only the function arguments were included, not the function name).